wasm/src/lib.rs

```diff
@@ -0,0 +1,636 @@
+use log::{Level, Record};
+use std::{
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+    os::raw::c_void,
+    panic,
+    panic::AssertUnwindSafe,
+    ptr,
+    time::{Duration, Instant},
+};
+use wasmer_runtime::{
+    error::{CallError, Error as WasmerError, RuntimeError},
+    Array, Ctx, Instance, WasmPtr,
+};
+
+pub trait Environment {
+    fn elapsed(&self) -> Result<Duration, Error>;
+
+    fn read_input(
```

Comment:
How will read_input relate to rustmatic::core::ProcessImage:: read_bit/read_byte/read_word/read_doubleword?

Reply:
I imagine `read_bit()` and friends would be implemented on top of `read_input()`, seeing as `read_input()` is more general.